dotnet/src/webdriver/Internal/Logging/ILogContext.cs

```diff
@@ -0,0 +1,85 @@
+    /// </summary>
+    public interface ILogContext : IDisposable
+    {
+        /// <summary>
+        /// Creates a new logging context.
+        /// </summary>
+        /// <returns>A new instance of <see cref="ILogContext"/>.</returns>
+        ILogContext CreateContext();
+
+        /// <summary>
+        /// Creates a new logging context with the specified minimum log level.
+        /// </summary>
+        /// <param name="minimumLevel">The minimum log level for the new context.</param>
+        /// <returns>A new instance of <see cref="ILogContext"/> with the specified minimum log level.</returns>
+        ILogContext CreateContext(LogEventLevel minimumLevel);
+
+        /// <summary>
+        /// Gets a logger for the specified type.
+        /// </summary>
+        /// <typeparam name="T">The type for which to retrieve the logger.</typeparam>
+        /// <returns>An instance of <see cref="ILogger"/> for the specified type.</returns>
+        internal ILogger GetLogger<T>();
+
+        /// <summary>
+        /// Gets a logger for the specified type.
+        /// </summary>
+        /// <param name="type">The type for which to retrieve the logger.</param>
+        /// <returns>An instance of <see cref="ILogger"/> for the specified type.</returns>
+        internal ILogger GetLogger(Type type);
+
+        /// <summary>
+        /// Emits a log message using the specified logger, log level, and message.
+        /// </summary>
+        /// <param name="logger">The logger to emit the log message.</param>
+        /// <param name="level">The log level of the message.</param>
+        /// <param name="message">The log message.</param>
+        internal void EmitMessage(ILogger logger, LogEventLevel level, string message);
+
+        /// <summary>
+        /// Sets the minimum log level for the current context.
+        /// </summary>
+        /// <param name="level">The minimum log level.</param>
+        /// <returns>The current instance of <see cref="ILogContext"/> with the minimum log level set.</returns>
+        ILogContext SetLevel(LogEventLevel level);
+
+        /// <summary>
+        /// Sets the minimum log level for the specified type in the current context.
+        /// </summary>
+        /// <param name="issuer">The type for which to set the minimum log level.</param>
+        /// <param name="level">The minimum log level.</param>
+        /// <returns>The current instance of <see cref="ILogContext"/> with the minimum log level set for the specified type.</returns>
+        ILogContext SetLevel(Type issuer, LogEventLevel level);
+
+        /// <summary>
+        /// Adds a log handler to the current context.
+        /// </summary>
+        /// <param name="handler">The log handler to add.</param>
+        /// <returns>The current instance of <see cref="ILogContext"/> with the log handler added.</returns>
+        ILogContext AddHandler(ILogHandler handler);
```

Comment:
I will redesign it, we have to have a possibility to remove handlers (since `ConsoleLogHandler` is registered by default)

Reply:
Since this is just for Selenium, I think we're ok just supporting one? (e.g., setting a new handler replaces the old handler)

Reply:
We support many handlers. I need just expose some API to remove some of them.

Reply:
Resolved. Now we can use `Log.Handlers.Add(new Handler())`

Reply:
I guess I mean that for the other languages setting to output to a file doesn't also output to console

Reply:
Hidden stones! I (user) would like to manage all handlers.

```csharp
Log.Handlers.Clear().Add(new ConsoleLogHandler()).Add(new SqlLogHandler()).Add(new ReportPortalLogHandler());
```

Reply:
But I don't see a good use case for having multiple handlers, so it becomes just an extra step.

That said, I don't have strong feelings on it so long as it is well documented.

Reply:
Logging is event-based. Event may have subsriber**s**. Log handler is a subsriber.

Moreover reading docs for java impl:
![image](https://github.com/SeleniumHQ/selenium/assets/22616990/ae39401f-4f89-4616-8f61-31b13fd64c46)